### DISCUSSION TITLE
[FR] setting to remove toplevel `data` attribute (and `meta` attribute)

### DISCUSSION THREAD

**Discussion by jnachtigall** [ComfortablyCoding/strapi-plugin-transformer#38] [Ideas]
I am new to strapi. I use it with next.js. I already added you plugin (thanks!). However, I noticed that the toplevel `data` is still there, even with `removeDataKey: true`. So responses come as 
```json
{
  data: { ...my actual data... }
  meta: { always empty }
}
```

In my opinion it would be nice if the top level `data` and also the `meta` could also be removed.

In my next.js app with the redux-toolkit I always have to write an extra `data`:
```tsx
  const { data, error, isLoading } = useGetContentByIdQuery(id);

  return (
          <MyContent
              attr1={data.data.att1}
              attr2={data.data.att2}
            />
  );
```

So the typing of the response also has an unneeded `data`.

Now it is:
```ts
type MyContentType= {
  data: {
    id: number;
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    att1: string;
    att2: string;
  };
  meta: {};
};
```

But it should imho rather be:
```ts
type MyContentType= {
    id: number;
    createdAt: string;
    updatedAt: string;
    publishedAt: string;
    att1: string;
    att2: string;
};
```

**Comment by ComfortablyCoding** [comment]
The reason I left `data` property at top level as it's quite a common API structure to put all the data in a top level key. It prevents polluting the top level information and helps seperate it from other info like meta. 

I believe meta is currently empty for all cases aside from [pagination](https://docs.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest/sort-pagination.html#pagination). It will be utilized in the future for other features such as content versioning. So not something we just want to remove.

We can add a setting to remove the data key and shift everything top level but it will leave meta. We can add it as an additional setting and leave it up to the user with a warning of the repercussions of removing it. 

Alternatively instead of removing the `data` property we can add a setting to rename it if the property name `data` is the actual issue (e.g.  values so it would be data.values).

Let me know what you think!

**Comment by jnachtigall** [comment]
I see. Thanks for the explanation. 

Since I am new to Strapi maybe just move this to Discussions and see if there is some interest by other folks for such a setting. I might just remove the strapi-plugin-transformer again (to have an dependency less) and use https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#customizing-query-responses-with-transformresponse to transform the response on the frontend instead of on the strapi backend.
